collector/lib/HostHeuristics.cpp

```diff
@@ -57,9 +57,37 @@ class DockerDesktopHeuristic : public Heuristic {
   }
 };
 
+class S390XHeuristic : public Heuristic {
+ public:
+  // S390X does not support eBPF ealier than 4.18.0-348 (rhel8.5) so we switch to use corebpf
+  // instead.
+  void Process(HostInfo& host, const CollectorConfig& config, HostConfig* hconfig) const {
+    auto k = host.GetKernelVersion();
+    std::string os_id = host.GetOSID();
+
+    if (k.machine != "s390x" || config.GetCollectionMethod() == CollectionMethod::CORE_BPF) {
+      return;
+    }
+
+    // example release version: 4.18.0-305.88.1.el8_4.s390x
+    // build_id = 305
+    if (k.release.find(".el8") != std::string::npos) {
+      if (k.kernel == 4 && k.major == 18 && k.minor == 0) {
```

Comment:
I'm curious, why check for kernel version and build_id and not `el8_4`?

Reply:
followed same pattern as HostInfo.cpp isRHEL76 function.

Reply:
For RHEL 7, the kernel version is always `.el7.` whether it is 7.5 or 7.6 etc. But for RHEL 8+ the kernel build includes the minor number. I think checking for el8_4 would be more reliable.

Reply:
ok, I can change this, rebuild and retest if you feel like we didn't capture the appropriate releases.

To be clear, you want the release, kernel version and build id check to be removed in place of searching the release string for ".el8_4."  ?

Reply:
Yes, that is more straightforward. Especially after looking a little closer, the major build id doesn't change on patch releases.

| RHEL version | Kernel prefix | 
| -- | -- |
| RHEL 8.3 | 4.18.0-240 |
| RHEL 8.4 | 4.18.0-305 |
| RHEL 8.5 | 4.18.0-348 |
| RHEL 8.6 | 4.18.0-372 |

Reply:
pushed change for checking  "el8_4" in release string instead of checking on kernel version and build id.